main.go

```diff
@@ -43,28 +45,31 @@ func init() {
 		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
 	flag.BoolVar(&debug, "debug", false, "Debug logging")
 	flag.Parse()
-
-	_ = addonmgrv1alpha1.AddToScheme(scheme)
-	// +kubebuilder:scaffold:scheme
 }
 
 func main() {
 	ctrl.SetLogger(zap.New(zap.UseDevMode(debug)))
 
 	setupLog.Info(version.ToString())
 
+	nonCached := []client.Object{
```

Comment:
Why disabled cache for those objects? It is good to capture the reason with comment here.

Reply:
This is k8s controller pattern : tells the client that, if any cache is used, to bypass it for the given objects. The controllers are specifically interested in those objects.

Reply:
Nice catch, explaination added.